components/NumbersData/NumbersData.tsx

```diff
@@ -108,10 +111,15 @@ export default function NumbersPage({
               return (
                 <ListItem key={dataDate.toISOString()}>
                   <Link
-                    href={`/${citySlug}/${categoryData.slug}?d=${format(
-                      dataDate,
-                      "yyyy-MM-dd"
-                    )}`}
+                    href={{
+                      pathname: "/[city]/[category]",
+                      query: {
+                        city: query.city,
+                        category: query.category,
```

Comment:
Are we using `citySlug` or `categorySlug`? we could simplify props if getting this from `router.query`.